src/python/pants/option/global_options.py

```diff
@@ -780,7 +780,7 @@ class BootstrapOptions:
     )
     pants_bin_name = StrOption(
         advanced=True,
-        default="./pants",  # noqa: PANTSBIN
+        default="pants",  # noqa: PANTSBIN
```

Comment:
So thinking about this, I think we can salvage the plugin if we broaden its scope a little bit:
- Check for `./pants` and always warn
- Check for `pants` (lowercase) and warn
- Don't warn on `Pants`.
The idea being "Pants" is for prose, `pants` is for the name of the tool, which shouldn't be hardcoded.

Reply:
Because otherwise people hardcode `pants` in help strings for commands, which is unhelpful if you've changed the bin name 🙂

Reply:
Supporting changing the bin name, though we must honor it, is well past its prime IIUC. I know of no other binary that does this. Needless flexibility that leads to real bugs - lockfile headers 🤦

Reply:
Plenty of tools use argv0 for user messages. I assume this flag was to handle the fact that pants was run from a script so argv0 wasn't helpful

Reply:
I don't think so, but my scope on this is admittedly narrow. I know? - may be hallucinating - Toolchain was at one point using it to brand a `./tc` script in the transition from 1.x to 2.0 days or thereabouts.

Reply:
Yeah this was for two reasons:

1) because it was up in the air whether the default cli command would remain `./pants` (or `pants`) for branding reasons. We had thought about `./tc` back in 2019.
2) in case repos had custom wrapper scripts with other names 

But I think 2) is not really an important case here, since we wouldn't know about any special flags that script takes, or arg processing it does, so we'd be assuming that its args are the same as `pants` args, at which point a user can presumably do the trivial name mapping in their head.

And 1) is probably not going to happen at this point.

Reply:
I guess now we could use argv0, since we expect Pants to be invoked via a binary, and if you wrapper-script it that's fine but no one expects a tool to know about its wrapper scripts. We previously had to know because that was the only way to run Pants, but now, fortunately...

Reply:
Until then... Might be worth keeping our Ts crossed and our Is dotted

Reply:
Also I assume the argv0 is still `python` even with scie-pants. But that's easy to prove or disprove.